rule preprocess_variants:
    input:
        truth="truth.vcf",
        query="query.vcf",
        truth_regions="truth.bed",
        strats="stratifications.tsv",
        strat_dir = "strats_dir",
        genome = "genome.fasta",
        genome_index = "genome.fasta.fai"
    output:
        runinfo="results.runinfo.json",
        comparison_vcf="results.vcf.gz",
        metrics_summary="results.summary.csv",
        metrics_csv="results.csv",
        metrics_json="results.json.gz",
        roc_all_csv="results.roc.all.csv.gz",
        roc_indel = "results.roc.Locations.INDEL.csv.gz",
        roc_indel_pass = "results.roc.Locations.INDEL.PASS.csv.gz",
        roc_snp = "results.roc.Locations.SNP.csv.gz",
        roc_tsv="results.roc.tsv"
    params:
        engine="vcfeval",
        prefix="results",
        ## parameters such as -L to left-align variants
        extra=""
    log: "happy.log"
    threads: 2
    wrapper: "master/bio/hap.py/hap.py"
